query/security/ldap/src/main/java/org/codice/ddf/admin/ldap/discover/LdapTestAttributeMappings.java

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.ldap.discover;
+
+import static org.codice.ddf.admin.ldap.commons.LdapMessages.dnDoesNotExistError;
+import static org.codice.ddf.admin.ldap.commons.LdapMessages.mappingAttributeNotFoundError;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.codice.ddf.admin.api.DataType;
+import org.codice.ddf.admin.api.fields.FunctionField;
+import org.codice.ddf.admin.common.fields.base.function.TestFunctionField;
+import org.codice.ddf.admin.common.fields.base.scalar.BooleanField;
+import org.codice.ddf.admin.ldap.commons.LdapConnectionAttempt;
+import org.codice.ddf.admin.ldap.commons.LdapTestingUtils;
+import org.codice.ddf.admin.ldap.fields.LdapDistinguishedName;
+import org.codice.ddf.admin.ldap.fields.config.LdapSettingsField;
+import org.codice.ddf.admin.ldap.fields.connection.LdapBindUserInfo;
+import org.codice.ddf.admin.ldap.fields.connection.LdapConnectionField;
+import org.codice.ddf.admin.security.common.fields.wcpm.ClaimsMapEntry;
+import org.forgerock.opendj.ldap.Connection;
+import org.forgerock.opendj.ldap.Filter;
+import org.forgerock.opendj.ldap.SearchScope;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableList;
+
+public class LdapTestAttributeMappings extends TestFunctionField {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LdapTestAttributeMappings.class);
+
+    public static final String FIELD_NAME = "testAttributeFields";
+
+    public static final String DESCRIPTION =
+            "Tests whether the attributes mapped to claims exist on users beneath the user base.";
+
+    private LdapConnectionField conn;
+
+    private LdapBindUserInfo bindInfo;
+
+    private LdapSettingsField settings;
+
+    private LdapTestingUtils utils;
+
+    public LdapTestAttributeMappings() {
+        super(FIELD_NAME, DESCRIPTION);
+        conn = new LdapConnectionField().useDefaultRequired();
+        bindInfo = new LdapBindUserInfo().useDefaultRequired();
+        settings = new LdapSettingsField().useDefaultUserAttributes();
```

Comment:
We also need to make sure the list is populate. If you make the list of entries required, not just it's content, it will automatically validate for an empty list.